### PR TITLE
Smooth CommonEvents#onPlayerTick to avoid possible lag spark

### PR DESCRIPTION
**Pull request by DPurple**
Hope it helps

### CODE REVIEW

File: src/main/java/de/ellpeck/naturesaura/events/CommonEvents.java [hunk 1]
```diff
@@ -131,19 +136,25 @@ public void onWorldTick(TickEvent.WorldTickEvent event) {
...
             }
 
-            if (event.player.world.getGameTime() % 200 != 0)
+            if (event.player.world.getGameTime() % 200 != event.player.getEntityId() % 200)
+                return;
+
+            ServerPlayerEntity playerMp = (ServerPlayerEntity) event.player;
+            Advancement negativeAdvancement = playerMp.getServerWorld().getServer().getAdvancementManager().getAdvancement(NEGATIVE_ADVANCEMENT_KEY);
+            Advancement positiveAdvancement = playerMp.getServerWorld().getServer().getAdvancementManager().getAdvancement(POSITIVE_ADVANCEMENT_KEY);
+            if(negativeAdvancement != null && positiveAdvancement != null)
```
Comment (Ellpeck) [inline src/main/java/de/ellpeck/naturesaura/events/CommonEvents.java:150]: Why would you do this check here? Doesn't this just check if both of the advancements *exist*? The general advancement manager isn't bound to a specific player.

**Review by Ellpeck — CHANGES_REQUESTED** [review]

File: src/main/java/de/ellpeck/naturesaura/events/CommonEvents.java (hunk 1 above)
Comment (DPurple) [reply src/main/java/de/ellpeck/naturesaura/events/CommonEvents.java:150]: You are right

Comment (DPurple) [reply src/main/java/de/ellpeck/naturesaura/events/CommonEvents.java:150]: I wanna to check if player already have those advancement first
If player already have those advancement, triangulateAuraInArea() may be avoided

**Comment by Ellpeck** [comment]
Have you tested this? It looks good code-wise now.

**Comment by DPurple** [comment]
Sorry, I have it compiled and loaded, but I didn't tested it.
It shouldn't matter too much, everything work fine now

**Comment by Ellpeck** [comment]
This pull request became outdated with the 1.18 port. If you still want it to be merged, please update and test it. Thanks!

**Pull request closed without merging**